src/processor/operator/persistent/reader/csv/parallel_csv_reader.cpp

```diff
@@ -124,6 +124,7 @@ static offset_t tableFunc(TableFuncInput& input, TableFuncOutput& output) {
             auto result = parallelCSVLocalState->reader->continueBlock(outputChunk);
             outputChunk.state->selVector->selectedSize = result;
             if (result > 0) {
+                parallelCSVSharedState->fileOffset = parallelCSVLocalState->reader->getFileOffset();
```

Comment:
I feel that we could alternatively just use `blockIdx` to calculate progress. Each block guarantees to have size `PARALLEL_BLOCK_SIZE`. So you could calculate your fileOffset from `blockIdx`. If so, you don't need to write `parallelCSVSharedState->fileOffset` in multiple places.

Also I would try to enforce progress being calculated in shared state as long as it's doable.